To: vim_dev@googlegroups.com
Subject: Patch 7.3.159
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.159
Problem:    Using uninitialized pointer when out of memory.
Solution:   Check for NULL return value.
Files:	    src/mbyte.c


*** ../vim-7.3.158/src/mbyte.c	2010-10-27 13:37:39.000000000 +0200
--- src/mbyte.c	2011-04-11 14:17:39.000000000 +0200
***************
*** 4129,4135 ****
  	done = to - (char *)result;
      }
  
!     if (resultlenp != NULL)
  	*resultlenp = (int)(to - (char *)result);
      return result;
  }
--- 4129,4135 ----
  	done = to - (char *)result;
      }
  
!     if (resultlenp != NULL && result != NULL)
  	*resultlenp = (int)(to - (char *)result);
      return result;
  }
*** ../vim-7.3.158/src/version.c	2011-04-11 14:27:34.000000000 +0200
--- src/version.c	2011-04-11 14:28:08.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     159,
  /**/

-- 
I learned the customs and mannerisms of engineers by observing them, much the
way Jane Goodall learned about the great apes, but without the hassle of
grooming.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
